WcaOnRails/vendor/assets/javascripts/jquery.are-you-sure.js

```diff
@@ -68,13 +68,27 @@
       $field.data('ays-orig', getValue($field));
     };
 
-    var checkForm = function(evt) {
+    // Oftentimes, forms have some javascript that runs immediately when fields change.
+    // It makes sense to wait for those events to run (and possibly dirty or clean the form)
+    // before we check to see if the form is dirty. -JFLY
+    var checkForm = function(evt) {//JFLY
+      var that = this;
+      setTimeout(function() {//JFLY
+        checkFormNow.call(that, evt);//JFLY
+      }, 0);//JFLY
+    };//JFLY
+    var checkFormNow = function(evt) {
 
       var isFieldDirty = function($field) {
         var origValue = $field.data('ays-orig');
         if (undefined === origValue) {
           return false;
         }
+        if($field.is(":disabled")) {//JFLY
+          // Disabled fields cannot be dirty, as they won't
+          // submit anything to the server.
```

Comment:
1. Yeah, this is why rails's checkbox helper automatically generates a hidden input before every checkbox. See the "Gotcha" on http://apidock.com/rails/ActionView/Helpers/FormHelper/check_box.
2. That's true. So far my only reaction is "let's never do that!"

Despite the caveat of 2), I still do think that it makes sense to ignore disabled inputs everywhere for the purposes of notifying a user about a potential change they're not saving. I also think it's better to err on the side of *not* notifying the user rather than notifying the user when they didn't actually change something (which can cause a lot of confusion).